src/SSEConnection.js

```diff
@@ -0,0 +1,54 @@
+import AbstractConnection from "./AbstractConnection";
+import ConnectionEvent from "./ConnectionEvent";
+
+/**
+ * Server Sent Event Connections
+ */
+export default class SSEConnection extends AbstractConnection {
+  constructor(url, options = {}) {
+    super(url);
+    this.options = options;
+    this.source = null;
+    this.data = [];
+  }
+
+  open() {
+    const { withCredentials = false } = this.options;
```

Comment:
can you please move all the defaults into the construcvtor? again, see `XHRConnection` for reference